chapter10/voice-werewolf/werewolf/simulator.py

```
@@ -0,0 +1,441 @@
+                    "generationConfig": {"temperature": 0, "maxOutputTokens": 512},
+                },
+                ensure_ascii=False,
+            ).encode("utf-8")
+            key = os.environ["GEMINI_API_KEY"]
+            request = urllib.request.Request(
+                f"https://generativelanguage.googleapis.com/v1beta/models/{model}:generateContent?key={key}",
+                data=payload,
+                headers={"Content-Type": "application/json"},
+                method="POST",
+            )
+            with urllib.request.urlopen(request, timeout=90) as response:
```

Ruff (0.16.0)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)